### DISCUSSION TITLE
[Compilers] Does this mean zksolc-bin cannot be installed on WSL2?

### DISCUSSION THREAD

**Discussion by taurenshaman** [zkSync-Community-Hub/zksync-developers#138] [Compilers]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

0.8.19

### zksolc/zkvyper version?

1.3.14

### Explain the issue

```bash
Does this mean zksolc-bin cannot be installed on WSL2(Ubuntu 22.04)?
It looks that the lib cannot detect the Kind of the OS?

jerin@blade:/mnt/e/blockchain/zksync$ chmod a+x ./zksolc
jerin@blade:/mnt/e/blockchain/zksync$ ./zksolc
solc subprocess error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
jerin@blade:/mnt/e/blockchain/zksync$
```
```


### Link to your repo?

_No response_

**Comment by taurenshaman** [comment]
It looks that I should install this lib: https://docs.soliditylang.org/en/latest/installing-solidity.html#linux-packages
But after I installed `solc`, there are sill errors: 
```
jerin@blade:/mnt/e/blockchain/zksync$ ./zksolc
No input sources specified.
Error(s) found. Compilation aborted
```

**Reply by dutterbutter** [reply]
Hey @taurenshaman can you outline the steps you've taken please. Are you downloading the binaries from the release page or do you see this message when trying to compile contracts?

**Reply by taurenshaman** [reply]
The discord bot helped me resolved.

**Comment by taurenshaman** [comment]
1. install the original compiler: https://docs.soliditylang.org/en/latest/installing-solidity.html#linux-packages
2. download the zksync version - zksolc: https://github.com/matter-labs/zksolc-bin
3. move the `zksolc` file to project path
4. modify `hardhat.config.ts`, set the path to `compilerPath`:
```
const config: HardhatUserConfig = {
  zksolc: {
    version: "latest",
    settings: {
      compilerPath: "./node_modules/.bin/zksolc-linux-amd64-musl-v1.3.14",
      //viaIR: true,
      optimizer: {
        enabled: true,
        runs: 10000,
      }
    },
  },
```
